### PR TITLE
controller/usage: remove debug output

### PR DESCRIPTION
**Pull request by sttts**
### Description of your changes

That event is useless and even harms production systems as it congests the API, with that leads to throttling and the log is full of only these events.

**We should have a rule: every event that is 1:1 with reconciles is a blocker.**

I have: 

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- ~~[ ] Added or updated unit tests.~~
- ~~[ ] Added or updated e2e tests.~~
- ~~[ ] Linked a PR or a [docs tracking issue] to [document this change].~~
- [ ] Added `backport release-x.y` labels to auto-backport this PR.

Need help with this checklist? See the [cheat sheet].

[contribution process]: https://github.com/crossplane/crossplane/tree/master/contributing
[docs tracking issue]: https://github.com/crossplane/docs/issues/new
[document this change]: https://docs.crossplane.io/contribute/contribute
[cheat sheet]: https://github.com/crossplane/crossplane/tree/master/contributing#checklist-cheat-sheet

### CODE REVIEW

**Review by phisco — APPROVED** [review]
Sounds good to me

**Comment by sttts** [comment]
@phisco added another commit

**Review by phisco — APPROVED** [review]
still LGTM, @sttts. I wonder if we could avoid the DeepEqual, but I guess that's fine.

File: internal/controller/apiextensions/usage/reconciler.go [hunk 1]
```diff
@@ -406,11 +406,16 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	}
 
 	u.Status.SetConditions(xpv1.Available())
-	r.record.Event(u, event.Normal(reasonUsageConfigured, "Usage configured successfully."))
+
 	// We are only watching the Usage itself but not using or used resources.
 	// So, we need to reconcile the Usage periodically to check if the using
 	// or used resources are still there.
-	return reconcile.Result{RequeueAfter: r.pollInterval}, errors.Wrap(r.client.Status().Update(ctx, u), errUpdateStatus)
+	if !reflect.DeepEqual(u, orig) {
```
Comment (negz) [inline internal/controller/apiextensions/usage/reconciler.go:413]: Nit: Slight preference for `cmp.Equal` here: https://github.com/google/go-cmp

It probably won't make a difference in this specific use case, but I think we mostly use `cmp.Equal` in this codebase. It gives us more configurability on what equality means if/when we need it.

Comment (sttts) [reply internal/controller/apiextensions/usage/reconciler.go:413]: updated.

**Review by phisco — APPROVED** [review]

**Pull request merged**